compiler/il/OMRIL.cpp

```diff
@@ -779,6 +586,11 @@ OMR::IL::opCodeForDirectStore(TR::DataType dt)
 TR::ILOpCodes
 OMR::IL::opCodeForDirectWriteBarrier(TR::DataType dt)
    {
+   static_assert(TR::NumOMRTypes == (sizeof(OMR::IL::opCodesForDirectWriteBarrier) / sizeof(OMR::IL::opCodesForDirectWriteBarrier[0])),
+              "OMR::IL::opCodesForDirectWriteBarrier is not the correct size");
+
+   if (dt.isVector()) return TR::BadILOp;
```

Comment:
Similar question to what I asked for read barriers earlier.

Reply:
Please see the answer above.